src/views/templates/details/tabs/details/components/WorkloadProfile.tsx

```diff
@@ -0,0 +1,77 @@
+import * as React from 'react';
+import produce from 'immer';
+import useWorkloadProfile, {
+  workloadLabels,
+} from 'src/views/templates/list/hooks/useWorkloadProfile';
+
+import { ensurePath } from '@catalog/utils/WizardVMContext';
+import { TemplateModel, V1Template } from '@kubevirt-ui/kubevirt-api/console';
+import { useModal } from '@kubevirt-utils/components/ModalProvider/ModalProvider';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { VM_WORKLOAD_ANNOTATION } from '@kubevirt-utils/resources/vm';
+import { k8sUpdate } from '@openshift-console/dynamic-plugin-sdk';
+import {
+  Button,
+  DescriptionListDescription,
+  DescriptionListGroup,
+  DescriptionListTerm,
+} from '@patternfly/react-core';
+import { PencilAltIcon } from '@patternfly/react-icons';
+
+import { TemplateDetailsGridProps } from '../TemplateDetailsPage';
+
+import WorkloadProfileModal from './WorkloadProfileModal';
+
+const WorkloadProfile: React.FC<TemplateDetailsGridProps> = ({ template, editable }) => {
+  const { createModal } = useModal();
+  const { t } = useKubevirtTranslation();
+  const workload = useWorkloadProfile(template);
+
+  const updateWorkload = (updatedWorkload: string) => {
+    const updatedTemplate = produce<V1Template>(template, (templateDraft: V1Template) => {
+      if (!templateDraft.objects[0].spec.template.metadata.annotations)
+        ensurePath(templateDraft, 'objects[0].spec.template.metadata.annotations');
+
+      delete templateDraft.objects[0].spec.template.metadata.annotations[VM_WORKLOAD_ANNOTATION];
+
+      if (Object.values(workloadLabels).includes(updatedWorkload))
+        templateDraft.objects[0].spec.template.metadata.annotations[VM_WORKLOAD_ANNOTATION] =
+          Object.keys(workloadLabels).find((key) => workloadLabels[key] === updatedWorkload);
```

Comment:
btw:// vm may not be always object 0, best is to find the vm object
```
templateDraft.objects.find((obj) => { return obj.kind === 'VirtualMachine'; })
```

Reply:
we have a util function for it called `getTemplateVirtualMachineObject` under `utils` directory

Reply:
From what I was told, we want to display only **VM** Templates in the list, so I've implemented it this way, so we work only with such Templates here. Such Templates have only one object by design, so this check is unnecessary here. I am not a fan of "what if it changes sometimes in the future... " approach, because it can lead to producing lots of code that will never be really useful. So this is why I don't spend too much time here checking if the object is really a VM object.

Also I am not very happy using `getTemplateVirtualMachineObject` function, because it is a long name, IMHO it makes the code less readable, comparing to directly using `objects[0]`which is super clear, but I understand your point of view, folks. Thanks for that! 👍🏽

Reply:
> From what I was told, we want to display only VM Templates 

we should show by default only templates that have a vm object, but this is a templates list (all templates), and a template page (any template), we should be able to show all, and play nicely with all templates, including templates that more objects then just the vm object.

> Also I am not very happy using getTemplateVirtualMachineObject function

my vote is to do the find inline :-)
``` ts
const obj = templateDraft.objects.find((obj) => { return obj.kind === 'VirtualMachine'; });
// vs
const obj =  getTemplateVirtualMachineObject(templateDraft);
```

Reply:
I had discussion with the UX and there is no any reason to display other than VM templates.

Reply:
not displaying templates that don't have a vm object, does mean that you should crash if the template include other objects.

for example the template:
```
kind: Template
...
objects:
- kind: Service
  ...
- kind: VirtualMachine
  ...
```

is a template that include a VM object, gut the code you suggest will crash, it will take `object[0]` and assume it's a VM while it's a Service object.

Reply:
Here we are working with templates that were already filtered out from other than VM Templates.

Reply:
> Here we are working with templates that were already filtered out from other than VM Templates.

the example above it for a template the has a vm object, it should pass the current filtering

Reply:
I've updated the PR according to the suggestions.